### PR TITLE
Handle OpenAI /audio/speech route

### PR DESCRIPTION
**Pull request by narengogi**
**Description:**
- Added new route and handler for /audio/speech

**Related Issues:**
- #488

**Testing Done**
- [X] Generated audio in all the supported formats for /audio/speech
- [X] Error Response Handling for /audio/speech
- [X] Verified that the response stream is sent as is 
```
client = OpenAI(
    base_url="http://localhost:8787/v1",
    api_key="",
    default_headers=createHeaders(
        provider="openai",
    )
)

response = client.audio.with_streaming_response.speech.create(
    model="tts-1",
    voice="alloy",
    input=(
        "Once upon a time, in a quaint village nestled among rolling hills, there lived a curious young girl named Lily. "
        "She had always dreamed of adventure beyond the confines of her small town. One day, while exploring the attic "
        "of her grandmother's old house, Lily discovered a dusty, leather-bound book with strange symbols on its cover. "
        "As she opened it, a swirl of glittering dust escaped, and she found herself transported to a magical realm "
        "filled with talking animals, floating islands, and shimmering forests. Lily embarked on an epic journey, "
        "making friends with a wise owl, a mischievous fox, and a brave knight along the way. Together, they faced "
        "challenges, solved riddles, and ultimately saved the magical realm from an evil sorcerer who sought to plunge "
        "it into eternal darkness. Through her adventures, Lily learned the value of friendship, courage, and the power "
        "of believing in oneself. When she finally returned home, Lily realized that the greatest adventures often begin "
        "with a single step into the unknown, and that magic exists everywhere if you know where to look. "
        "As the years passed, Lily grew older, but she never forgot her magical adventure. She became a storyteller, "
        "sharing tales of wonder and imagination with children in her village. Her stories inspired a new generation "
        "of dreamers and adventurers, each eager to find their own magical realms. And though she never found that "
        "mysterious book again, Lily knew that the real magic had always been within her heart, waiting to be discovered."
    ),
)

with response as streaming_response:
    streaming_response.stream_to_file("output.mp3")
```

**TODO**
- [X] Test .pcm format in /audio/speech route
```
used `ffplay  -autoexit -f s16le  ~/Documents/generated-speech/fox.pcm` to play file

### CODE REVIEW

File: src/handlers/createSpeechHandler.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      request,
+      requestHeaders,
+      'createSpeech',
+      'POST',
+      'config'
+    );
+
+    return tryTargetsResponse;
+  } catch (err: any) {
+    console.log('chatCompletion error', err.message);
```
Comment (VisargD) [inline src/handlers/createSpeechHandler.ts:32]: Please fix this log

File: src/providers/openai/createSpeech.ts
Comment (VisargD) [inline src/providers/openai/createSpeech.ts:1]: speech handler should also be added for azure-openai

**Comment by narengogi** [comment]
Changes now in https://github.com/Portkey-AI/gateway/pull/538

**Comment by CYzhr** [comment]
Hi! I built AICostMonitor - an open-source LLM cost tracking tool. Would love to discuss potential integration or partnership. My paypal: paypal.com/paypalme/Cyzhr

**Pull request closed without merging**